docs/user_guide/adversarial.rst

```diff
@@ -0,0 +1,281 @@
+
+.. _adversarial:
+
+Adversarial Learning
+--------------------
+
+.. currentmodule:: fairlearn.adversarial
+
+Fairlearn provides an implementation of the adversarial
+learning method presented in [#4]_.
+We assume we have data :math:`X, A, Y`, where we want to
+predict :math:`Y` from :math:`X` while being fair with respect to :math:`A`
+based on some fairness measure. We firstly create predictor and adversary neural
+networks with learnable weights :math:`W` and :math:`U` respectively. Without
+considering fairness yet, this typical supervised-learning problem aims to
+minimize the predictor loss :math:`L_P`. Now, to improve fairness, we not
+only want to minimize the predictor loss, but we want to decrease the
+adversary's ability to predict the sensitive features from the predictor's
+predictions. Suppose the adversary has loss term :math:`L_A`, then the paper
+trains the predictor with gradient:
+
+.. math::
+    \nabla_W L_P - \text{proj}_{\nabla_W L_A} \nabla_W L_P - \alpha \nabla_W L_A
+
+If this model converges properly,
+the adversary will attain a loss equal to the entropy, so the adversary
+can not
+predict the sensitive features from the predictions.
+Moreover, this model can be trained for either *demographic parity* or
```

Comment:
Side question: can this be extended to just do True Positive Rate Parity or False Positive Rate Parity? I suppose what I'm getting at is that I suspect those are just the _implemented_ constraints?

Reply:
More constraints can be implemented, such as equality of opportunity, but for TPR or FPR I am not quite sure what objective the adversary should then have. Perhaps it is possible.